includes/shipping/free-shipping/class-wc-shipping-free-shipping.php

```diff
@@ -143,8 +151,10 @@ public function is_available( $package ) {
 			$total = WC()->cart->get_displayed_subtotal();
 
 			if ( WC()->cart->display_prices_including_tax() ) {
-				$total = round( $total - ( WC()->cart->get_discount_total() + WC()->cart->get_discount_tax() ), wc_get_price_decimals() );
-			} else {
+				$total = round( $total - WC()->cart->get_discount_tax(), wc_get_price_decimals() );
+			}
+
+			if ( 'no' === $this->ignore_discounts ) {
```

Comment:
Maybe this new code can generate problem with rounding values twice.
Better only round those values after this block, like:

```php
if ( WC()->cart->display_prices_including_tax() ) {
	$total = $total - WC()->cart->get_discount_tax();
}

if ( 'no' === $this->ignore_discounts ) {
	$total = $total - WC()->cart->get_discount_total();
}

$total = round( $total, wc_get_price_decimals() );
```

Reply:
Thank you, i perfectly understand the problem and i have pushed the fix as you suggested.